### PR TITLE
[#129] Make management entrypoint tests shared

### LINKED ISSUE #129: Make management entrypoints' tests shared between Ligo and Lorentz
# Clarification and motivation

We have some shared tests for the Ligo and Lorentz version for most entrypoints, but not for the "management" ones implemented for #100 (see [this discussion](https://github.com/tqtezos/baseDAO/pull/111#discussion_r557406874)).

The difficult part was mostly due to the different errors between the Ligo and Lorentz version, now that #106 has been completed however, we should try to move as many tests to be shared to reduce duplication.

# Acceptance criteria

As many tests as possible for management entrypoints are `ShareTest` and used in both the Lorentz and Ligo version of the contract.

### PR DESCRIPTION
**Pull request by rinn7e**
## Description

Problem: Management tests between Lorentz and LIGO version are very
similar. We should create shared tests to use between them instead to
reduce duplication.

Solution: Make management tests shared between Lorentz and LIGO.

## Related issue(s)

Resolves #129 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by rinn7e** [comment]
Hope to get your review on this as well @pasqu4le .

File: ligo/haskell/test/Test/Ligo/BaseDAO/Management.hs [hunk 1]
```diff
@@ -245,34 +144,4 @@ test_BaseDAO_Management =
...
-expectMigrated
-  :: (MonadNettest caps base m)
-  => Address -> m a -> m ()
-expectMigrated addr = expectCustomError #mIGRATED addr
-
-expectForbiddenXTZ
-  :: (MonadNettest caps base m)
-  => m a -> m ()
-expectForbiddenXTZ = expectCustomError_ #fORBIDDEN_XTZ
+      ! defaults
```
Comment (pasqu4le) [inline ligo/haskell/test/Test/Ligo/BaseDAO/Management.hs:147]: Missing an end-of-file newline

File: src/BaseDAO/ShareTest/Management.hs [hunk 2]
```diff
@@ -0,0 +1,296 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+--
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+-- For all the incomplete list pattern matches in the calls to with
+-- withOriginatedFn func
```
Comment (pasqu4le) [inline src/BaseDAO/ShareTest/Management.hs:6]: Nitpick here as well.
```suggestion
-- For all the incomplete list pattern matches in the calls to the
-- `withOriginatedFn` function
```

File: test/Test/BaseDAO/Management.hs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 --
 {-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
 -- For all the incomplete list pattern matches in the calls to with
--- withOriginated func
+-- withOriginated initialStorage func
```
Comment (pasqu4le) [inline test/Test/BaseDAO/Management.hs:6]: Same nitpick as above

File: test/Test/BaseDAO/Management.hs [hunk 4]
```diff
@@ -12,16 +12,39 @@ module Test.BaseDAO.Management
...
 import Named (defaults, (!))
-import Test.Common
--- import Tezos.Core (unsafeMkMutez)
-import Util.Named
+
+import qualified Lorentz.Contracts.BaseDAO as DAO
+import Lorentz.Contracts.BaseDAO.Types
+import BaseDAO.ShareTest.Management
+
+-- | Function that originates the contract and also make a bunch of
```
Comment (pasqu4le) [inline test/Test/BaseDAO/Management.hs:24]:
```suggestion
-- | Function that originates the contract and also makes a bunch of
```

**Review by pasqu4le — COMMENTED** [review]
Only minor nitpicks, otherwise LGTM

File: test/Test/BaseDAO/Management.hs (hunk 4 above)
Comment (rinn7e) [reply test/Test/BaseDAO/Management.hs]: I've made all the changes you suggested now.

**Review by pasqu4le — COMMENTED** [review]
One more thing: in the issue title and description, as well as in the commit message you spelled `managment` which seems to be and incorrect form of `management`, please fix them.

Feel free to squash the commits as well

**Comment by rinn7e** [comment]
> One more thing: in the issue title and description, as well as in the commit message you spelled `managment` which seems to be and incorrect form of `management`, please fix them.
> 
> Feel free to squash the commits as well

Yeah, right, I didn't check it properly, I've squashed and fixed the typo now, you can check this, Pasquale.

**Review by pasqu4le — APPROVED** [review]

**Pull request merged**